Building failed for Cadence target

### 🐛 Describe the bug

I used cmd as below to create executorch components for cadence target:
`rm -rf cmake-out-cadence

CXXFLAGS="-fno-exceptions -fno-rtti" cmake \
  -DCMAKE_TOOLCHAIN_FILE=./backends/cadence/cadence.cmake \
  -DCMAKE_INSTALL_PREFIX=./cmake-out-cadence \
  -DCMAKE_BUILD_TYPE=Release \
  -DBUILD_SHARED_LIBS=OFF \
  -DEXECUTORCH_BUILD_CADENCE=ON \
  -DEXECUTORCH_BUILD_EXECUTOR_RUNNER=OFF \
  -DEXECUTORCH_BUILD_EXTENSION_RUNNER_UTIL=ON \
  -DEXECUTORCH_BUILD_PTHREADPOOL=OFF \
  -DEXECUTORCH_BUILD_CPUINFO=OFF \
  -DEXECUTORCH_BUILD_PORTABLE_OPS=ON \
  -DEXECUTORCH_BUILD_KERNELS_LLM=OFF \
  -DEXECUTORCH_ENABLE_LOGGING=ON \
  -DEXECUTORCH_ENABLE_PROGRAM_VERIFICATION=ON \
  -DEXECUTORCH_USE_DL=OFF \
  -DEXECUTORCH_NNLIB_OPT=ON \
  -DEXECUTORCH_ENABLE_EVENT_TRACER=OFF \
  -DHAVE_FNMATCH_H=OFF \
  -DFLATCC_ALLOW_WERROR=OFF \
  -B cmake-out-cadence`

**And, error rised:
CMake Error at backends/cadence/hifi/kernels/CMakeLists.txt:8 (add_library):
  Cannot find source file:

    /mnt/d/workspace/executorch/backends/cadence/hifi/third-party/nnlib/xa_nn_broadcast_32.c**

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm
  .ccm .cxxm .c++m .h .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90
  .f95 .f03 .hip .ispc


CMake Error at backends/cadence/hifi/kernels/CMakeLists.txt:8 (add_library):
  No SOURCES given to target: cadence_kernels

Then, I checked source from newest main branch commit, executorch/backends/cadence/hifi/third-party/nnlib/xa_nn_broadcast_32.c is not found indeed. 
So, should I change **backends/cadence/hifi/kernels/CMakeLists.txt** or use some other method to resolve it?

### Versions

PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04.3) 11.4.0
Clang version: Could not collect
CMake version: version 3.31.10
Libc version: glibc-2.3

Python version: 3.10.8 (Wed Jun 14 15:29:07 CST 2023) [Graal, GraalVM CE, Java 17.0.7] (32-bit runtime)
Python platform: Linux-6.6.87.2-microsoft-standard-WSL2-x86_64-with-glibc2.3
Is CUDA available: N/A
CUDA runtime version: 12.5.40
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4060 Laptop GPU
Nvidia driver version: 555.97
cuDNN version: Could not collect
Is XPU available: N/A
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A
Caching allocator config: N/A

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        39 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               32
On-line CPU(s) list:                  0-31
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Core(TM) i9-14900HX
CPU family:                           6
Model:                                183
Thread(s) per core:                   2
Core(s) per socket:                   16
Socket(s):                            1
Stepping:                             1
BogoMIPS:                             4838.39
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology tsc_reliable nonstop_tsc cpuid tsc_known_freq pni pclmulqdq vmx ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves avx_vnni vnmi umip waitpkg gfni vaes vpclmulqdq rdpid movdiri movdir64b fsrm md_clear serialize flush_l1d arch_capabilities
Virtualization:                       VT-x
Hypervisor vendor:                    Microsoft
Virtualization type:                  full
L1d cache:                            768 KiB (16 instances)
L1i cache:                            512 KiB (16 instances)
L2 cache:                             32 MiB (16 instances)
L3 cache:                             36 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-31
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Mitigation; Clear Register File
Vulnerability Retbleed:               Mitigation; Enhanced IBRS
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.2.6
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-nccl-cu12==2.27.5
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] torch==2.11.0.dev20260215+cpu
[pip3] triton==3.6.0
[conda] nvidia-cublas-cu12          12.8.4.1                pypi_0                   pypi
[conda] nvidia-cuda-cupti-cu12      12.8.90                 pypi_0                   pypi
[conda] nvidia-cuda-nvrtc-cu12      12.8.93                 pypi_0                   pypi
[conda] nvidia-cuda-runtime-cu12    12.8.90                 pypi_0                   pypi
[conda] nvidia-cudnn-cu12           9.10.2.21               pypi_0                   pypi
[conda] nvidia-cufft-cu12           11.3.3.83               pypi_0                   pypi
[conda] nvidia-curand-cu12          10.3.9.90               pypi_0                   pypi
[conda] nvidia-cusolver-cu12        11.7.3.90               pypi_0                   pypi
[conda] nvidia-cusparse-cu12        12.5.8.93               pypi_0                   pypi
[conda] nvidia-cusparselt-cu12      0.7.1                   pypi_0                   pypi
[conda] nvidia-nccl-cu12            2.27.5                  pypi_0                   pypi
[conda] nvidia-nvjitlink-cu12       12.8.93                 pypi_0                   pypi
[conda] nvidia-nvtx-cu12            12.8.90                 pypi_0                   pypi
[conda] torch                       2.11.0.dev20260215+cpu  pypi_0                   pypi
[conda] triton                      3.6.0                   pypi_0                   pypi

--- a/backends/cadence/aot/functions_hifi.yaml
+++ b/backends/cadence/aot/functions_hifi.yaml
@@ -192,6 +192,11 @@
     - arg_meta: null
       kernel_name: impl::HiFi::minimum_out
 
+- op: native_layer_norm.out
+  kernels:
+    - arg_meta: null
+      kernel_name: torch::executor::native_layer_norm_out
+
 - op: mm.out
   kernels:
     - arg_meta: null
@@ -499,6 +504,11 @@
     - arg_meta: null
       kernel_name: impl::HiFi::quantized_relu_asym8u_asym8u_per_tensor_out
 
+- func: cadence::quantized_max_pool2d_nchw.out(Tensor input, int[] kernel_size, int[] stride, int[] padding, int[] dilation, bool ceil_mode, *, Tensor(a!) out) -> Tensor(a!)
+  kernels:
+    - arg_meta: null
+      kernel_name: impl::generic::quantized_max_pool2d_nchw_out
+
 - func: cadence::quantized_max_pool2d_nhwc.out(Tensor input, int[] kernel_size, int[] stride, int[] padding, int[] dilation, bool ceil_mode, *, Tensor(a!) out) -> Tensor(a!)
   kernels:
     - arg_meta: null
@@ -552,9 +562,9 @@
 - func: cadence::quantized_conv1d_ncl.per_tensor_out(Tensor input, Tensor weight, Tensor bias, int[] stride, SymInt[] padding, int[] dilation, int groups, int input_zero_point, int weight_zero_point, float bias_scale, float out_scale, int out_zero_point, int out_multiplier, int out_shift, *, Tensor(a!) out) -> Tensor(a!)
   kernels:
     - arg_meta: null
-      kernel_name: impl::HiFi::native::quantized_conv1d_ncl_per_tensor_out
+      kernel_name: impl::HiFi::quantized_conv1d_ncl_per_tensor_out
 
 - func: cadence::quantized_conv1d_nlc.per_tensor_out(Tensor input, Tensor weight, Tensor bias, int[] stride, SymInt[] padding, int[] dilation, int groups, int input_zero_point, int weight_zero_point, float bias_scale, float out_scale, int out_zero_point, int out_multiplier, int out_shift, *, Tensor(a!) out) -> Tensor(a!)
   kernels:
     - arg_meta: null
-      kernel_name: impl::HiFi::native::quantized_conv1d_nlc_per_tensor_out
+      kernel_name: impl::HiFi::quantized_conv1d_nlc_per_tensor_out

--- a/backends/cadence/hifi/operators/CMakeLists.txt
+++ b/backends/cadence/hifi/operators/CMakeLists.txt
@@ -59,6 +59,8 @@ set(_aten_ops__srcs
     "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_clone.cpp"
     "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_gelu.cpp"
     "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_max_pool2d_with_indices.cpp"
+    "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_native_layer_norm.cpp"
+    "${EXECUTORCH_ROOT}/kernels/portable/cpu/util/normalization_ops_util.cpp"
     "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_to_copy.cpp"
     "${EXECUTORCH_ROOT}/kernels/portable/cpu/pattern/unary_ufunc_realhbbf16_to_floathbf16.cpp"
     "${EXECUTORCH_ROOT}/kernels/portable/cpu/util/activation_ops_util.cpp"
@@ -99,7 +101,10 @@ set(_generic_kernels__srcs
 # impl::generic::native implementations called by HiFi operators.
 set(_generic_custom_ops__srcs
     "${EXECUTORCH_ROOT}/backends/cadence/generic/operators/op_quantized_linear.cpp"
+    "${EXECUTORCH_ROOT}/backends/cadence/generic/operators/op_quantized_conv1d_ncl.cpp"
+    "${EXECUTORCH_ROOT}/backends/cadence/generic/operators/op_quantized_conv1d_nlc.cpp"
     "${EXECUTORCH_ROOT}/backends/cadence/generic/operators/op_quantized_conv2d.cpp"
+    "${EXECUTORCH_ROOT}/backends/cadence/generic/operators/op_quantized_max_pool2d.cpp"
     "${EXECUTORCH_ROOT}/backends/cadence/generic/operators/op_quantized_matmul.cpp"
 )
 
@@ -140,6 +145,7 @@ add_library(
   "op_quantized_conv2d_nhwc_depthwise_asym8uxsym8u_asym8u_per_tensor_out.cpp"
   "op_quantized_conv2d_nhwc_dilated_asym8sxsym8s_asym8s_per_tensor_out.cpp"
   "op_quantized_conv2d_nhwc_dilated_asym8uxsym8u_asym8u_per_tensor_out.cpp"
+  "op_quantized_max_pool2d_nhwc.cpp"
   "op_quantized_fully_connected_out.cpp"
   "op_quantized_fully_connected_asym8sxasym8s_asym8s_per_tensor_out.cpp"
   "op_quantized_fully_connected_asym8uxasym8u_asym8u_per_tensor_out.cpp"

--- a/backends/cadence/install_requirements.sh
+++ b/backends/cadence/install_requirements.sh
@@ -30,7 +30,7 @@ if [ $STATUS -ne 0 ]; then
     exit 1
 fi
 
-git checkout 102944a6f76a0de4d81adc431f3f132f517aa87f
+git checkout 64d73d729797388b0e346794d7a2e408374a74de
 
 
 ## Fusion G3

